Updating branch refs/heads/xfce-4.8
         to a1c52f8b557b9ebd0e2b89cac70cd6cd9346ae5b (commit)
       from d5048bb865362b2265c467b3f730f4bb6707b97c (commit)

commit a1c52f8b557b9ebd0e2b89cac70cd6cd9346ae5b
Author: Nick Schermer <nick@xfce.org>
Date:   Sat Jun 4 12:46:26 2011 +0200

    Fix preferred applications handling (bug #7140).
    
    I broke the writing in mimeapps.list after removing the
    mimetypes keys from the desktop files in commit 2e3744b.
    So restore them again as X-XFCE-MimeType.
    
    (cherry picked from commit 41b00df8433e832f9cee43d596dc16bf8d56971c)

 exo-helper/exo-helper.c              |    2 +-
 exo-open/exo-file-manager.desktop.in |    1 +
 exo-open/exo-mail-reader.desktop.in  |    1 +
 exo-open/exo-web-browser.desktop.in  |    2 +-
 4 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/exo-helper/exo-helper.c b/exo-helper/exo-helper.c
index 3fe7229..161477b 100644
--- a/exo-helper/exo-helper.c
+++ b/exo-helper/exo-helper.c
@@ -731,7 +731,7 @@ exo_helper_database_set_default (ExoHelperDatabase *database,
   if (G_UNLIKELY (desktop_file != NULL))
     {
       xfce_rc_set_group (desktop_file, "Desktop Entry");
-      mimetypes = xfce_rc_read_list_entry (desktop_file, "MimeType", ";");
+      mimetypes = xfce_rc_read_list_entry (desktop_file, "X-XFCE-MimeType", ";");
       if (mimetypes != NULL)
         {
           xfce_rc_set_group (rc, "Added Associations");
diff --git a/exo-open/exo-file-manager.desktop.in b/exo-open/exo-file-manager.desktop.in
index ca4aedf..3d7653e 100644
--- a/exo-open/exo-file-manager.desktop.in
+++ b/exo-open/exo-file-manager.desktop.in
@@ -7,5 +7,6 @@ StartupNotify=true
 Terminal=false
 Categories=Utility;X-XFCE;X-Xfce-Toplevel;
 OnlyShowIn=XFCE;
+X-XFCE-MimeType=x-scheme-handler/file;x-scheme-handler/trash;
 _Name=File Manager
 _Comment=Browse the file system
diff --git a/exo-open/exo-mail-reader.desktop.in b/exo-open/exo-mail-reader.desktop.in
index 541de0d..511fff7 100644
--- a/exo-open/exo-mail-reader.desktop.in
+++ b/exo-open/exo-mail-reader.desktop.in
@@ -7,5 +7,6 @@ StartupNotify=false
 Terminal=false
 Categories=Network;X-XFCE;X-Xfce-Toplevel;
 OnlyShowIn=XFCE;
+X-XFCE-MimeType=x-scheme-handler/mailto;
 _Name=Mail Reader
 _Comment=Read your email
diff --git a/exo-open/exo-web-browser.desktop.in b/exo-open/exo-web-browser.desktop.in
index 65870e5..49f31eb 100644
--- a/exo-open/exo-web-browser.desktop.in
+++ b/exo-open/exo-web-browser.desktop.in
@@ -7,6 +7,6 @@ StartupNotify=false
 Terminal=false
 Categories=Network;X-XFCE;X-Xfce-Toplevel;
 OnlyShowIn=XFCE;
+X-XFCE-MimeType=x-scheme-handler/http;x-scheme-handler/https;
 _Name=Web Browser
 _Comment=Browse the web
-
_______________________________________________
Xfce4-commits mailing list
Xfce4-commits@xfce.org
http://foo-projects.org/mailman/listinfo/xfce4-commits
